lib/product-import.coffee

```diff
@@ -135,24 +150,27 @@ class ProductImport
           @logger.warn "Filtering out #{filteredProductsLength} product(s) which do not have SKU"
           @_summary.productsWithMissingSKU += filteredProductsLength
 
-        skus = @_extractUniqueSkus(productsToProcess)
-        if skus.length then @_getExistingProductsForSkus(skus) else []
-      .then (queriedEntries) =>
         if (@variantReassignmentOptions.enabled)
           @logger.debug 'execute reassignment process'
+
           reassignmentService = new Reassignment(@client, @logger,
-            @variantReassignmentOptions.retainExistingData)
+             @variantReassignmentOptions.retainExistingData)
           reassignmentService.execute(productsToProcess, @_cache.productType)
-          .then((statistics) =>
-            @_summary.variantReassignment = statistics
-            if (statistics.processed > 0)
-              skus = @_extractUniqueSkus(productsToProcess)
-              if skus.length then @_getExistingProductsForSkus(skus) else queriedEntries
-            else
-              return queriedEntries
+          .then((res) =>
+            @_summary.variantReassignment = res.statistics
+
+            # if there are products which failed during reassignment, remove them from processing
+            if(res.failedSkus.length)
+              @logger.warn(
+                "Removing #{res.failedSkus} skus from processing due to a reassignment error"
+              )
+              productsToProcess = @_filterOutProductsBySkus(productsToProcess, res.failedSkus)
+
+            # fetch existing products - they should be reassigned now
+            @_getExistingProductsForSkus(@_extractUniqueSkus(productsToProcess))
```

Comment:
You shoud put back this condition. If there was no reassignment, it should not refetch the products, but just use what has been passed. It's quite a big performance improvement.

```if (statistics.processed > 0)```

Reply:
check this https://github.com/sphereio/sphere-product-import/pull/121/files/e26174e30881cb5370bbaa91867d7c0e2f8947a3#diff-d7b70bb31c9da8f845222653763558c7L139 I removed fetching on the top and moved it here because we did not use `queriedEntries` anywhere in the code during reassignment .. but I will change it a little bit

Reply:
Please check my change here 49a57a0 - the process now works like this:
 - get productsToProcess (productDrafts)
 - call reassignment and filter out any failed products from productsToProcess
 - take productsToProcess, extract skus and fetch existing products
 - continue with the original `product-import` process